src/logging.c

```diff
@@ -195,7 +203,7 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
     }
     if (RCUTILS_GET_ENV_ZERO == retval || RCUTILS_GET_ENV_ONE == retval) {
       int mode = retval == RCUTILS_GET_ENV_ZERO ? _IONBF : _IOLBF;
-      if (setvbuf(g_output_stream, NULL, mode, 0) != 0) {
+      if (setvbuf(g_output_stream, NULL, mode, RCUTILS_LOGGING_STREAM_BUFFER_SIZE) != 0) {
```

Comment:
Actually, reading through the glibc code, I think this is buggy in another way as well.

If I'm understanding the glibc code properly, anytime you pass 0 for the size, its equivalent to specifying `_IONBF`; the `mode` argument is completely ignored in that case.  That means we are basically ignoring what the user is asking for in `RCUTILS_LOGGING_BUFFERED_STREAM` and always setting the stream to unbuffered.

So with that said, I'm wondering if the fix here is to just simple change this line to:

```
if (setvbuf(g_output_stream, NULL, mode, BUFSIZ) != 0) {
```

If we are setting `_IONBF`, that final size argument will be ignored.  If we are setting `_IOLBF`, then it will be doing the correct thing on both Windows and Linux.

@brawner What do you think?

Reply:
That looks like the right call. Added new commit. The glibc suggests BUFSIZ is good for most situations. If a better choice is required, it can be found using fstat.

https://www.gnu.org/software/libc/manual/html_node/Controlling-Buffering.html